common/sky/altitude.py

```diff
@@ -0,0 +1,99 @@
+import numpy as np
+import numpy.typing as npt
+import astropy.units as u
+from astropy.coordinates import Angle, Longitude
+from typing import Tuple, Union
+
+class Altitude:
+
+    def above(dec: Union[Angle, float, npt.NDArray[float]],
```

Comment:
Since we're doing this `Union[T, U, npt.NDArray[U]]` over and over again, why not do, at the beginning of the file:

```
from typing import TypeVar
T = TypeVar('T')
U = TypeVar('U')
ScalarOrArray = Union[T, U, npt.NDArray[U]]
AngleParam = ScalarOrArray[Angle, float]
```

and then we can change all these repetitive parameter types to:
```
dec: AngleParam
```
etc.

(I think that should work... I tried it in a Python console.)
```